lib/node_modules/@stdlib/blas/ext/base/dnanasumors/src/main.c

```diff
@@ -17,45 +17,56 @@
 */
 
 #include "stdlib/blas/ext/base/dnanasumors.h"
+#include "stdlib/strided/base/stride2offset.h"
 #include "stdlib/math/base/assert/is_nan.h"
 #include "stdlib/math/base/special/abs.h"
-#include <stdint.h>
+#include "stdlib/blas/base/shared.h"
 
 /**
 * Computes the sum of absolute values (L1 norm) of double-precision floating-point strided array elements, ignoring `NaN` values and using ordinary recursive summation.
 *
-* @param N       number of indexed elements
-* @param X       input array
-* @param stride  stride length
-* @return        output value
+* @param N        number of indexed elements
+* @param X        input array
+* @param strideX  stride length
+* @return         output value
 */
-double stdlib_strided_dnanasumors( const int64_t N, const double *X, const int64_t stride ) {
+double API_SUFFIX(stdlib_strided_dnanasumors)( const CBLAS_INT N, const double *X, const CBLAS_INT strideX ) {
+	CBLAS_INT ox = stdlib_strided_stride2offset( N, strideX );
+	return API_SUFFIX(stdlib_strided_dnanasumors_ndarray)( N, X, strideX, ox );
+}
+
+/**
+* Computes the sum of absolute values (L1 norm) of double-precision floating-point strided array elements, ignoring `NaN` values and using ordinary recursive summation and alternative indexing semantics.
+*
+* @param N        number of indexed elements
+* @param X        input array
+* @param strideX  index increment
+* @param offsetX  starting index
+* @return         output value
+*/
+double API_SUFFIX(stdlib_strided_dnanasumors_ndarray)( const CBLAS_INT N, const double *X, const CBLAS_INT strideX, const CBLAS_INT offsetX ) {
 	double sum;
-	int64_t ix;
-	int64_t i;
+	CBLAS_INT ix;
+	CBLAS_INT i;
 	double v;
 
 	if ( N <= 0 ) {
 		return 0.0;
 	}
-	if ( N == 1 || stride == 0 ) {
+	if ( N == 1 || strideX == 0 ) {
```

Comment:
Same comment as the JS version. I think this conditional should be removed in its entirety. It's possible, but I don't know for sure, that, e.g., pairwise summation needed a non-zero stride and so this branch is included for parity, but I don't know. @headlessNode Would you mind doing a little bit of code archeology and R&D?

Reply:
Sounds reasonable to me.

Reply:
Thanks for investigating!